pom.xml

```diff
@@ -911,6 +913,26 @@
                     </limit>
                   </limits>
                 </rule>
+                <rule>
+                  <element>CLASS</element>
+                  <includes>
+                    <include>
+                      com.puppycrawl.tools.checkstyle.Main
+                    </include>
+                  </includes>
+                  <limits>
+                    <limit>
+                      <counter>LINE</counter>
+                      <value>COVEREDRATIO</value>
+                      <minimum>0.99</minimum>
```

Comment:
@romani ping
So the only way is to drop line coverage for the class?

Reply:
@rnveach some workarounds are possible. We can keep the coverage at 100% for the `Main` class.
By moving the call to `System.exit` to some other class (this class will be not 100% covered).
Or we can mark this code with special annotation to turn JaCoCo coverage off for one method.
Both ways are tricky, it is better to count the coverage as is.

Reply:
> Or we can mark this code with special annotation to turn JaCoCo coverage off for one method.

The only way I see to do this is using `@generated` annotation, this is not ideal.

> By moving the call to System.exit to some other class (this class will be not 100% covered).

This seems like a hack to me.

I think it is better that we stay with coverage specification in `pom.xml`, it is the probably the best option out of the three given.

Reply:
On second thought, could we cover `System.exit()` call by execution of `Main#main` with  `TestUtil#getResultWithLimitedResources` ?

Reply:
> On second thought, could we cover `System.exit()` call by execution of `Main#main` with `TestUtil#getResultWithLimitedResources` ?

`System.exit()` just forces VM to quit. No coverage probes will be touched.
This can be hacked with a custom `SecurityManager`, but in this case this method will throw an exception.
The problem is that
```java
System.exit(1); // Coverage collected for this  line
// Coverage NOT COLLECTED for this  line 
} // Coverage collected for this  line, the program ends here
```

The workaround we using for JDK < 16 is about to simply mock `System.exit`. But now this is not possible.

Reply:
Finally solved with a helper inner class. Now JaCoCo coverage for the `Main` class is 100%